### PR TITLE
Adding amplitude events for update school dialog

### PR DESCRIPTION
**Pull request by hannahbergam**
Adds four amplitude events for the school info update dialog, as spelled out in product spec linked below, plus one added from PR feedback.

I also updated the a11y of this component to make it tab navigable by removing the __useDeprecatedTag. I had to update the styling of the update button: see before and after below. 
Shoutout to @wilkie who translated this Storybook last year, which made development here much much faster! So fun to see the fruits of that labor!

To be clear: this is only the first portion of Amplitude work: there is a dialog that follows this if a user presses 'update' that is not yet instrumented. 

Screenshot 2024-02-12 at 6 57 03 PM
Screenshot 2024-02-12 at 6 57 10 PM
Screenshot 2024-02-12 at 6 57 27 PM
Screenshot 2024-02-13 at 9 32 39 AM

Before:
Screenshot 2024-02-12 at 6 58 52 PM

After (excuse the active blue outline on the 'x'- this was just me testing out tabnav):
Screenshot 2024-02-12 at 7 03 53 PM

(This is how it looks RTL:)
Screenshot 2024-02-12 at 7 05 50 PM

## Links

- spec: https://docs.google.com/document/d/1P1RVtGDy4XbIs0t5hnHC3Li6ZDsCnjwMnOPMQqPXNyk/edit?pli=1#bookmark=id.y2xg1jxvnayj
- jira ticket: https://codedotorg.atlassian.net/browse/ACQ-1466

## PR Checklist:

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: apps/src/lib/util/AnalyticsConstants.js [hunk 1]
```diff
@@ -11,6 +11,12 @@ const EVENTS = {
   ABANDON_SECTION_SETUP_SIGN_IN_EVENT: 'Abandon Section Setup Sign In',
   TEACHER_LOGIN_EVENT: 'Teacher Login',
 
+  // School Association
+  // Update School Info Dialog
+  UPDATE_SCHOOL_INFO_DIALOG_SHOWN: 'Update School Info Dialog Shown',
+  CONFIRM_SCHOOL_CLICKED: 'Confirm School Clicked',
+  UPDATE_SCHOOL_CLICKED: 'Update School Clicked',
+
```
Comment (bethanyaconnor) [inline apps/src/lib/util/AnalyticsConstants.js:20]: I know this isn't in the product spec but curious if it's work having an `UPDATE_SCHOOL_INFO_DIALOG_CLOSED` event? Or do we think we can figure that out by calculating the number of users that didn't confirm or update?

**Review by bethanyaconnor — APPROVED** [review]
Looks good!

File: apps/src/lib/util/AnalyticsConstants.js (hunk 1 above)
Comment (dmcavoy) [reply apps/src/lib/util/AnalyticsConstants.js:20]: That sounds like a good idea!

Comment (hannahbergam) [reply apps/src/lib/util/AnalyticsConstants.js:20]: Added!

**Review by bethanyaconnor — APPROVED** [review]

**Pull request merged**